### PR TITLE
ESP32-C3: Add support for running apps

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

Add support for running Tock applications

### Testing Strategy

Run apps

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

**Review by hudson-ayers — APPROVED** [review]

**Review by lschuermann — DISMISSED** [review]

File: boards/esp32-c3-devkitM-1/layout.ld [hunk 1]
```diff
@@ -1,8 +1,8 @@
 MEMORY
 {
   rom (rx)  : ORIGIN = 0x40380000, LENGTH = 0x30000
-  ram (rwx) : ORIGIN = 0x403B0000, LENGTH = 0x30000
```
Comment (bradjc) [inline boards/esp32-c3-devkitM-1/layout.ld:4]: How did this work?!?

**Review by bradjc — APPROVED** [review]

File: boards/esp32-c3-devkitM-1/layout.ld (hunk 1 above)
Comment (lschuermann) [reply boards/esp32-c3-devkitM-1/layout.ld:4]: Based on the datasheet it appears to me as both of these sections are located in SRAM. Perhaps you made the same mistake as I did and read the `B` as an `8`?

Comment (alistair23) [reply boards/esp32-c3-devkitM-1/layout.ld:4]: It's inside the SRAM memory space

**Review by ppannuto — APPROVED** [review]
bors d+

File: boards/esp32-c3-devkitM-1/layout.ld (hunk 1 above)
Comment (bradjc) [reply boards/esp32-c3-devkitM-1/layout.ld:4]: So if you unplug the board the apps go away?

Comment (alistair23) [reply boards/esp32-c3-devkitM-1/layout.ld:4]: No, it's confusing. Tock and apps are loaded into flash. The boot ROM then loads that flash into RAM and executes it.

So the read-only regions of Tock need to be in RAM as that is where the boot ROM jumps to. In theory we could store the apps in flash, but the ROM seems to do all sorts of strange things so I haven't been able to get that to works. Plus I don't think that would get us much as Tock can't use fragmented memory addresses

**Pull request merged**